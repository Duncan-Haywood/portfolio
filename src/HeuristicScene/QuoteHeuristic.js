import * as helpers from './HelperFunctions.js'
import React from 'react'

/**
 * Heuristic Quote
 */
export function Quote( props ) {
	let darkColor = helpers.getDarkColor({schemeIndex: props.currentHeuristic});
	let lightColor = helpers.getLightColor({schemeIndex: props.currentHeuristic});
	quoteComponent = <div className="heuristic__quote"
			style={{
				backgroundColor: darkColor,
				color: lightColor,
			}}>
			{ props.heuristic }
		</div>

	return ( quoteComponent);
}
Quote.defaultProps = { 
	currentHeuristic: undefined, heuristic: undefined //, getDarkColor: getDarkColor, getLightColor: getLightColor, colorSchemes: colorSchemes 
}