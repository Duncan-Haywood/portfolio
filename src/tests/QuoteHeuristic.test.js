import React from "react";
import { configure, shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { Quote } from '../HeuristicScene/QuoteHeuristic.js'
import Adapter from "enzyme-adapter-react-16";
import {HEURISTICS, INTRO} from '../HeuristicScene/Constants.js'

configure({ adapter: new Adapter() });




const tableQuoteTest = [
	{currentHeuristic: 1, heuristic: HEURISTICS[0]},
	{currentHeuristic: "", heuristic: INTRO},
	{currentHeuristic: 2, heuristic: HEURISTICS[1]},
	{currentHeuristic: HEURISTICS.length, heuristic: HEURISTICS[HEURISTICS.length]},
]


describe("Quote component", () => {
	// let className = "heuristic__primary";
	// let currentHeuristic = 1;
	test.each(tableQuoteTest)("renders shallowly", (props) => {
		shallow(<Quote currentHeuristic={ props.currentHeuristic } 
					heuristic={ props.heuristic } />)
	})
	
	test.todo("renders shallowly with a variety of classNames and currentHeuristics")
	test.todo("renders shallowly")
	test.todo("renders fully")
})
