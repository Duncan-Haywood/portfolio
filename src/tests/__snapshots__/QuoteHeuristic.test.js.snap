// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote component Quote snapshots Snapshot for: { currentHeuristicIndex: '', heuristic: [Object] } 1`] = `
<div
  className="heuristic__quote"
  style={
    Object {
      "backgroundColor": "#aa3f00",
      "color": "#faffc4",
    }
  }
>
  <p>
    <strong>
      Welcome!
    </strong>
     
    Thank you so much for visiting; It means a lot to me that you're engaging in my passions.
  </p>
  <p>
    This is one of my first large scale projects that I have hosted. And, if you'd like to check, I'll be sharing more art projects here over time.
  </p>
  <p>
     Sincerely,
     
    <a
      href="https://www.linkedin.com/in/duncanhaywood"
    >
      Duncan Haywood
    </a>
    .
  </p>
</div>
`;

exports[`Quote component Quote snapshots Snapshot for: { currentHeuristicIndex: 1, heuristic: [Object] } 1`] = `
<div
  className="heuristic__quote"
  style={
    Object {
      "backgroundColor": "#2d7a00",
      "color": "#efffc2",
    }
  }
>
  <p>
    The background and navigation of this web app was built by the amazing 
     
    <a
      href="http://moc.co"
    >
      Joen
    </a>
     
     
    who shared this 
     
    <a
      href="https://github.com/jasmussen/turtleshell"
    >
      project
    </a>
     
     
    as open source.
  </p>
  <p>
    The quotes in the following pages are also created by Joen. Much thanks to him.
  </p>
</div>
`;

exports[`Quote component Quote snapshots Snapshot for: { currentHeuristicIndex: 2, heuristic: [Object] } 1`] = `
<div
  className="heuristic__quote"
  style={
    Object {
      "backgroundColor": "#ff9f60",
      "color": "#00113c",
    }
  }
>
  <p>
    So far I've added the music and sound effects to this site. I aim to wrap different art based machine learning projects and incorporate them into this project next.
  </p>
</div>
`;

exports[`Quote component Quote snapshots Snapshot for: { currentHeuristicIndex: 36, heuristic: [Object] } 1`] = `
<div
  className="heuristic__quote"
  style={
    Object {
      "backgroundColor": "#aa3f00",
      "color": "#faffc4",
    }
  }
>
  <p>
    Invest in trees. The long term ROI could be your way of life.
  </p>
</div>
`;
